src/components/Api.js

```diff
@@ -0,0 +1,129 @@
+class Api {
+  constructor(options) {
+    this.options = options;
+  }
+
+  // Common method for handling fetch requests
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return res.json().then((data) => Promise.reject(data));
+  }
+
+  // Common method for handling fetch errors
+  _handleError(err) {
+    console.error(`Error: ${err.message}\nURL: ${err.url}`);
+  }
+
+  // Get user information
+  getUserInfo() {
+    return fetch(`${this.options.baseUrl}/users/me`, {
+      headers: {
+        authorization: this.options.headers.authorization,
+      },
+    })
+      .then(this._handleResponse)
+      .catch(this._handleError);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.